src/Microsoft.Health.Dicom.Blob/Features/Export/AzureBlobExportSink.cs

```diff
@@ -67,31 +68,41 @@ public async Task<bool> CopyAsync(ReadResult value, CancellationToken cancellati
         InstanceMetadata i = value.Instance;
         try
         {
-            using Stream sourceStream = await _source.GetStreamingFileAsync(i.VersionedInstanceIdentifier.Version,
-                i.VersionedInstanceIdentifier.Partition, i.InstanceProperties.fileProperties, cancellationToken);
+            using Stream sourceStream = await _source.GetStreamingFileAsync(
+                i.VersionedInstanceIdentifier.Version,
+                i.VersionedInstanceIdentifier.Partition,
+                i.InstanceProperties.FileProperties,
+                cancellationToken);
+
             BlobClient destBlob = _dest.GetBlobClient(_output.GetFilePath(i.VersionedInstanceIdentifier));
-            await destBlob.UploadAsync(sourceStream, new BlobUploadOptions { TransferOptions = _blobOptions.Upload },
+            await destBlob.UploadAsync(
+                sourceStream,
+                new BlobUploadOptions { TransferOptions = _blobOptions.Upload },
                 cancellationToken);
             return true;
         }
-        catch (DataStoreException ex) when (ex.IsExternal)
-        {
-            throw;
-        }
-        catch (DataStoreRequestFailedException ex) when (ex.IsExternal)
+        catch (Exception ex) when (ShouldContinue(ex))
         {
-            return LogFailure(i, ex);
-        }
-        catch (Exception ex) when (ex is not RequestFailedException rfe || rfe.Status < 400 || rfe.Status >= 500) // Do not include client errors
-        {
-            return LogFailure(i, ex);
+            CopyFailure?.Invoke(this, new CopyFailureEventArgs(i.VersionedInstanceIdentifier, ex));
+            EnqueueError(DicomIdentifier.ForInstance(i.VersionedInstanceIdentifier), ex.Message);
+            return false;
         }
     }
 
-    private bool LogFailure(InstanceMetadata i, Exception ex)
+    private static bool ShouldContinue(Exception ex)
     {
-        CopyFailure?.Invoke(this, new CopyFailureEventArgs(i.VersionedInstanceIdentifier, ex));
-        EnqueueError(DicomIdentifier.ForInstance(i.VersionedInstanceIdentifier), ex.Message);
+        // don't continue if the data has been modified in the source as it is likely an issue that won't be fixed by retrying
+        if (ex is DataStoreRequestFailedException dsrfe && dsrfe.IsExternal && dsrfe.ResponseCode == (int)HttpStatusCode.PreconditionFailed)
+            return false;
```

Comment:
This is backwards: returning `true` means we log the issue in the export error log and continue processing the other files. I think we can simply say:

```csharp
catch (DataStoreRequestFailedException e) when (e.IsExternal && e.ResponseCode == (int)HttpStatusCode.PreconditionFailed))
```